imap_processing/ultra/l1b/ultra_l1b_annotated.py

```diff
@@ -61,6 +74,12 @@ def get_particle_velocity(
 
     # Apply Compton-Getting.
     # Particle velocity in the DPS frame wrt to the heliosphere
-    particle_velocity_heliosphere = spacecraft_velocity + particle_velocity_spacecraft
+    particle_velocity_heliosphere = (
```

Comment:
I'm concerned about this calculation. I think that `spacecraft_velocity` should always be 0 in the `IMAP_DPS` frame because that frame moves with the spacecraft. I think that you need to get the spacecraft velocity in the heliosphere frame (`ECLIPJ2000`) and add that to the particle velocity expressed in the heliosphere frame. Something like:
```
particle_vel_helio = frame_transform(
    et=time,
    position=instrument_velocity,
    from_frame=instrument_frame,
    to_frame=SpiceFrame.ECLIP2000
)
# Spacecraft velocity in the pointing (DPS) frame wrt heliosphere.
state = imap_state(time, ref_frame=SpiceFrame.ECLIPJ2000)
# Extract the velocity part of the state vector
spacecraft_velocity = state[:, 3:6]
particle_vel_compton_corrected = spacecraft_velocity + particle_velocity_dps_spacecraft
```

Reply:
Nevermind, `imap_state` defaults to using the sun as the observing body.